### PR TITLE
Creating Report (#6)

### PR DESCRIPTION
**Pull request by kegliitis**
Creation of report and some code structure reformation

### CODE REVIEW

File: Sep3DataTier/Repository/DatabaseContext.cs [hunk 1]
```diff
@@ -34,9 +34,9 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         modelBuilder.Entity<Location>().HasKey(location => location.Id);
         modelBuilder.Entity<Model.Report>().HasKey(report => report.Id);
         modelBuilder.Entity<ApplicationUser>().HasKey(user => user.Id);
-        modelBuilder.Entity<ApplicationUser>().HasIndex(u => u.Email).IsUnique();
-        modelBuilder.Entity<IdentityRole>().HasData(GenerateIdentityRole("User"));
-        modelBuilder.Entity<IdentityRole>().HasData(GenerateIdentityRole("Admin"));
+        modelBuilder.Entity<ApplicationUser>().HasIndex(user => user.Email).IsUnique();
+        // modelBuilder.Entity<IdentityRole>().HasData(GenerateIdentityRole("User"));
+        // modelBuilder.Entity<IdentityRole>().HasData(GenerateIdentityRole("Admin"));
```
Comment (apeTrackunit) [inline Sep3DataTier/Repository/DatabaseContext.cs:39]: Is there a reason for commenting that?

File: Sep3DataTier/Services/ReportService.cs [hunk 2]
```diff
@@ -50,4 +60,50 @@ public override async Task<ReportList> GetReports(ReportFilter request, ServerCa
...
         });
     }
+
+    public override async Task<ReportObject> CreateReport(CreateReportObject request, ServerCallContext context)
+    {
+        Location location = await locationDao.CreateLocationAsync(new Location
+        {
+            Latitude = request.Location.Latitude,
+            Longitude = request.Location.Longitude,
+            // Size = BitConverter.GetBytes(request.Location.Size)[1]
```
Comment (apeTrackunit) [inline Sep3DataTier/Services/ReportService.cs:70]: Can be removed?

**Review by apeTrackunit — CHANGES_REQUESTED** [review]

File: Sep3DataTier/Repository/DatabaseContext.cs (hunk 1 above)
Comment (kegliitis) [reply Sep3DataTier/Repository/DatabaseContext.cs:39]: For some reason, migrations weren't working for me when it wasn't commented out.

File: Sep3DataTier/Services/ReportService.cs (hunk 2 above)
Comment (kegliitis) [reply Sep3DataTier/Services/ReportService.cs:70]: Yes

**Review by JcullenNZ — APPROVED** [review]

**Pull request merged**